### ISSUE TITLE
Add a column for types in the output table

### ISSUE THREAD

**Issue by skyzyx** [segmentio/terraform-docs#269]
**Is your feature request related to a problem? Please describe.**

In Terraform 0.12, output blocks can refer to _objects_ (otherwise known as _types_). To use a contrived example:

```hcl
data "aws_arn" "db_instance" {
  arn = "arn:aws:rds:eu-west-1:123456789012:db:mysql-db"
}

output "aws_arn_db_instance" {
    value = data.aws_arn.db_instance
}
```

Then…

```hcl
resource whatever "whatever" {
    input = module.whatever.aws_arn_db_instance.arn
}
```

When I'm working with one of these modules, and I want to access a property (attribute) on the output, I end up needing to look at the code to figure out that it's a `aws_arn` data resource — then, I need to look it up on the Terraform website to see which properties (attributes) are available on that object/type.

It adds a lot of cognitive overhead.

**Describe the solution you'd like**

Add a third column to the outputs table (i.e., after the name and description) which displays the _type_ of the output. It could be a "simple" type like `string`, `number`, `map`, `list`… or it could be a "complex" provider object like `aws_arn`, `newrelic_alert_channel`, `pagerduty_user`, etc.

Even better would be to auto-link to the Terraform documentation for that object (I have an algorithm I'd be happy to share for doing this for my own projects).

**Comment by khos2ow** [comment]
This is a very good suggestion, and I can see the practical use for it considering in Terraform 0.12 whole module or resource can be _outputed_.

But I cannot think of way to implement this at the moment, because of the way output resources are being discovered by `terraform-config-inspect` and the fact that the actual values and types of the outputs are not known unless explicitly generated with `terraform output -json` (even I'm not entirely sure about the types though).

I'm gonna explore this further to see if I can come up with something for it.

**Comment by skyzyx** [comment]
So, I just read https://github.com/hashicorp/terraform-config-inspect/issues/21 which sounds like you forked from upstream. I did just put in a feature request on that repo to see if Hashicorp is willing to adopt/support this request, but we'll see.

**Comment by khos2ow** [comment]
Yes we in-tree forked the project tin [internal/tfconfig](https://github.com/segmentio/terraform-docs/tree/master/internal/tfconfig), because the upstream `terraform-config-inspect` is considered feature complete and they don't accept any PR.

So in reality we have some flexibility to update `internal/tfconfig` but the way terraform-config-inspect works is the limitation here. I mean there's no (easy) way to extract these information without changing the code extensively.

Although to be clear I'm not rejecting the idea completely, I'm just saying it might be harder than expected.

**Comment by nitrocode** [comment]
I'd like a similar `types` column for the resources table too. It's confusing when you see the following:

```
aws_iam_role
aws_iam_role
```

and then you have to mouse over each one to see which one to click on to get the resource docs.

**Comment by skyzyx** [comment]
Being able to document outputs is probably a separate issue from being able to document which type of resource one of the internal type is.

But it's one I'd 👍🏻.

With the recent update of `terraform-docs` being able to generate a resource list/table, it _almost_ replaces the Python script I wrote which is identical except that mine also says which type. For example:

* [aws_iam_role](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_role) (data source)
* [aws_iam_role](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role) (resource)
